File: 프로그래머스/3/77886. 110 옮기기/110 옮기기.js
function solution(s) {
    const answer = [];
    for(const str of s) {
        const stack = [];
        let count = 0;
        for(const char of str) {
            stack.push(char);
            if(stack[stack.length - 3] === '1' &&
              stack[stack.length - 2] === '1' &&
              stack[stack.length - 1] === '0') {
                count++;
                stack.splice(-3);
            }
        }
        const idx = stack.lastIndexOf('0');
        if(idx === -1) {
            answer.push('110'.repeat(count) + stack.join(''));
        } else {
            stack.splice(idx + 1, 0, '110'.repeat(count));
            answer.push(stack.join(''));
        }
    }
    return answer;
}

File: 프로그래머스/3/77886. 110 옮기기/110 옮기기.py
def solution(s):
    answer = []
    for char in s:
        stack = []
        count = 0
        for c in char:
            stack.append(c)
            while len(stack) >= 3 and stack[-3:] == ['1', '1', '0']:
                count += 1
                for _ in range(3):
                    stack.pop()
        if '0' not in stack:
            answer.append('110' * count + "".join(stack))
        else:
            idx = len(stack) - 1 - stack[::-1].index('0')
            answer.append("".join(stack[:idx + 1]) + '110' * count + "".join(stack[idx + 1:]))
    return answer
                
                